packages/message/src/DeadLetterMessage.ts

```diff
@@ -24,26 +57,25 @@ type UnwrapDeadLetterCumulusMessageReturnType = (
  * @returns {Object} the cumulus message or nearest available object
  */
 export const unwrapDeadLetterCumulusMessage = async (
-  messageBody: UnwrapDeadLetterCumulusMessageReturnType
-): Promise<UnwrapDeadLetterCumulusMessageReturnType> => {
+  messageBody: UnwrapDeadLetterCumulusMessageInputType
+): Promise<CumulusMessage | UnwrapDeadLetterCumulusMessageInputType> => {
   try {
-    if ('cumulus_meta' in messageBody) {
-      return messageBody;
-    }
-    if ('Body' in messageBody || 'body' in messageBody) {
+    if (isSQSRecordLike(messageBody)) {
       // AWS.SQS.Message/SQS.Record case
       const unwrappedMessageBody = parseSQSMessageBody(
-        messageBody
-      ) as CumulusMessage;
+        messageBody as SQSRecord | AWS.SQS.Message
+      ) as StepFunctionEventBridgeEvent;
       return await unwrapDeadLetterCumulusMessage(unwrappedMessageBody);
     }
-    if (!('detail' in messageBody)) {
-      // Non-typed catchall
-      return messageBody;
+    if (isEventBridgeLike(messageBody)) {
+      return await getCumulusMessageFromExecutionEvent(
+        messageBody as StepFunctionEventBridgeEvent
+      );
+    }
+    if (isCumulusMessageLike(messageBody)) {
+      return messageBody as CumulusMessage;
     }
-    // StepFunctionEventBridgeEvent case
-    const unwrappedMessageBody = await getCumulusMessageFromExecutionEvent(messageBody);
-    return await unwrapDeadLetterCumulusMessage(unwrappedMessageBody);
+    throw new TypeError('DeadLetter CumulusMessage in unrecognized format');
   } catch (error) {
```

Comment:
1. so this error is an unwrap error, so doesn't need to be added as part of the DLA message?
2. I assume this is the case, but wanted to check - if we encounter this unwrap error are we going to see the full message in the logs so we can understand why the unwrap error was encountered?

Reply:
this shouldn't appear in DLA, this is meant to remove duplicate code and clarify something. as previously written:
 - if message is in an unrecognized format (found by first not entering "if 'cumulus_meta' in obj" and "if 'Body' in obj", and then asking "is 'detail' not in obj?)
 - if an error is thrown somewhere in there
 
 this instead checks for truth in "if 'detail' in obj" and therefore treats it appropriately, and if it doens't fit into any of the buckets, treats that as an error. it should be functionally unchanged, but easier to read what's happening

Reply:
this one is hard to say in text, I'd be happy to pair to clarify

Reply:
re-looking at this, I think we never got around to discussing question #2. Maybe we can talk about it.

Reply:
adding in failed messageBody